src/vs/workbench/contrib/terminal/browser/terminalTabbedView.ts

```diff
@@ -89,7 +90,8 @@ export class TerminalTabbedView extends Disposable {
 		this._instanceMenu = this._register(menuService.createMenu(MenuId.TerminalContext, contextKeyService));
 		// this._dropdownMenu = this._register(menuService.createMenu(MenuId.TerminalTabsContext, contextKeyService));
 
-		this._tabsWidget = this._instantiationService.createInstance(TerminalTabsWidget, this._terminalTabTree);
+		const terminalLabels = this._register(this._instantiationService.createInstance(ResourceLabels, DEFAULT_LABELS_CONTAINER));
+		this._tabsWidget = this._instantiationService.createInstance(TerminalTabsWidget, this._terminalTabTree, terminalLabels);
```

Comment:
this.register(createInstance (terminaltabswidget)

create terminalLabels within the tabsWidget